libraries/botbuilder-azure/src/doOnce.ts

```diff
@@ -7,13 +7,20 @@
  */
 
 export class DoOnce<T> {
-    private task: Promise<T>;
+    private task: {
+        [dbAndContainerKey: string]: Promise<T>;
+    } = {};
 
-    public waitFor(fn: () => Promise<T>): Promise<T> {
-        if (!this.task) {
-            this.task = fn();
+    public waitFor(dbAndContainerKey: string, fn: () => Promise<T>): Promise<T> {
+        if (!this.task[dbAndContainerKey]) {
+            // Surround in a try/catch so that if container creation fails, it can retry and succeed later.
+            try {
```

Comment:
Should this be something like:
```
            try {
                this.task[dbAndContainerKey] = fn();
            } catch (err) {
                delete this.task[dbAndContainerKey];
                console.error(err);
            }
```
So the Task is removed, and future callers can re-add it.